subscription.go

```diff
@@ -257,58 +306,75 @@ type handlerFunc func(data []byte, err error) error
 
 // Subscription stores the subscription declaration and its state
 type Subscription struct {
+	id      string
```

Comment:
What is the difference between id and the key, so to have both per subscription?

Reply:
Oh, I see but looking up the subscription either by the key or the id seems complicating, no?

Reply:
We can separate them into 2 functions. However, I just unify the search function into a single function. We still need to find the subscription by id internally but the user only has the original id before resetting.